src/components/keyboard/index.jsx

```diff
@@ -19,7 +19,14 @@ export default function Keyboard({ checkGuess, gameOver }) {
             return;
         }
         event.preventDefault();
-        setInput(event.target.value.toLowerCase());
+        if (event.target.value.match(/[^a-z]/g)) {
```

Comment:
You can simplify this a lot. It's important to remember that React will keep the value of the input in sync with whatever `input` is; so we just can just rewrite `input` after every input.  What you've got here works, but since you're changing `event.target.value` outside of the scope of what React is tracking.  What'd we want to do is something more like:

```javascript
    function onChange(event) {
        if (gameOver) {
            return;
        }
        event.preventDefault();
        const val = event.target.value.replace(/[^a-zA-Z]/g, '').toLowerCase();
        if (val.length <= word.length) {
            setInput(val);
            updateActive(val);
        }
    }
```

On every input, we take the value, remove anything that's non-alpha, and lowercase what's left, shove it back into `input` and `active`.

Reply:
Ohh okay still trying to figure out how to chain things like this along together but this makes perfect sense. Whenever the real keyboard gets added we can just remove the `setInput` from here and only use active.